scripts/stream_throughput.py

```diff
@@ -4,14 +4,59 @@
 
 Description: Provides a mechanism for measuring Stabilizer stream data throughput.
 """
+import argparse
 import socket
 import collections
 import struct
 import time
 import logging
 
-# Representation of a single UDP packet transmitted by Stabilizer.
-Packet = collections.namedtuple('Packet', ['index', 'adc', 'dac'])
+# Representation of a single data batch transmitted by Stabilizer.
+Packet = collections.namedtuple('Packet', ['index', 'data'])
+
+# The magic header half-word at the start of each packet.
+MAGIC_HEADER = 0x057B
+
+# The struct format of the header.
+HEADER_FORMAT = '<HBBI'
+
+# All supported formats by this reception script.
+#
+# The items in this dict are functions that will be provided the sampel batch size and will return
+# the struct deserialization code to unpack a single batch.
+FORMAT = {
+    1: lambda batch_size: f'<{batch_size}H{batch_size}H{batch_size}H{batch_size}H'
+}
+
+def parse_packet(buf):
+    """ Attempt to parse packets from the received buffer. """
+    # Attempt to parse a block from the buffer.
+    if len(buf) < struct.calcsize(HEADER_FORMAT):
+        return []
+
+    # Parse out the packet header
+    magic, format_id, batch_size, sequence_number = struct.unpack_from(HEADER_FORMAT, buf)
+    buf = buf[struct.calcsize(HEADER_FORMAT):]
+
+    if magic != MAGIC_HEADER:
+        logging.warning('Encountered bad magic header: %s', hex(magic))
+        return []
+
+    if format_id not in FORMAT:
+        raise Exception(f'Unknown format specifier: {format_id}')
+
+    frame_format = FORMAT[format_id](batch_size)
+
+    batch_count = int(len(buf) / struct.calcsize(frame_format))
+
+    packets = []
+    for offset in range(batch_count):
+        data = struct.unpack_from(frame_format, buf)
+        buf = buf[struct.calcsize(frame_format):]
+        packets.append(Packet(sequence_number + offset, data))
```

Comment:
```suggestion
        yield Packet(sequence_number + offset, data)
```

and remove the empty lists in `return []` and the `return packets` below.